### PR TITLE
Flag tests failing as expected (by version)

### PR DESCRIPTION
**Pull request by martica**
This attempts to mark all the outstanding test failures as expected to allow other changes to be submitted with proper test coverage and avoid continued tests from rotting.

## PR Checklist:

- [ ] All new features have been tested
- [ ] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: tests/structures/test_datamodel.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # coding=utf-8
-from ..utils import TranspileTestCase
+from ..utils import TranspileTestCase, expected_failing_versions
 from unittest import expectedFailure
 
 class ClassBinaryOpsDataModelTests(TranspileTestCase):
```
Comment (brutusthebee) [inline tests/structures/test_datamodel.py]: At column 1: (E302) expected 2 blank lines, found 1

File: tests/structures/test_datamodel.py [hunk 2]
```diff
@@ -511,6 +511,7 @@ def __ior__(self, other):
 
 @expectedFailure
```
Comment (brutusthebee) [inline tests/structures/test_datamodel.py]: At column 1: (E302) expected 2 blank lines, found 1

File: tests/modules/test_stdlib.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from ..utils import TranspileTestCase
+from ..utils import TranspileTestCase, expected_failing_versions
 
 import unittest
```
Comment (brutusthebee) [inline tests/modules/test_stdlib.py]: At column 1: (F401) 'unittest' imported but unused

File: tests/datatypes/test_str.py [hunk 4]
```diff
@@ -728,6 +728,7 @@ class FormatTests(TranspileTestCase):
                 print('Done.')
                 """
 
+        @expected_failing_versions(['3.5', '3.6'])
         @transforms(
             js_bool = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 5]
```diff
@@ -728,6 +728,7 @@ class FormatTests(TranspileTestCase):
...
             decimal = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 6]
```diff
@@ -841,6 +842,7 @@ def test_precision(self, js_cleaner, py_cleaner):
 
             self.assertCodeExecution(tests, js_cleaner = js_cleaner, py_cleaner = py_cleaner)
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 55: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 57: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 80: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 82: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 7]
```diff
@@ -841,6 +842,7 @@ def test_precision(self, js_cleaner, py_cleaner):
...
 
+        @expected_failing_versions(['3.6'])
         @transforms(
             js_bool = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 8]
```diff
@@ -841,6 +842,7 @@ def test_precision(self, js_cleaner, py_cleaner):
...
             decimal = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 9]
```diff
@@ -870,6 +872,7 @@ def test_left_adjust(self, js_cleaner, py_cleaner):
 
             self.assertCodeExecution(tests, js_cleaner = js_cleaner, py_cleaner = py_cleaner)
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 55: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 57: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 80: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 82: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 10]
```diff
@@ -870,6 +872,7 @@ def test_left_adjust(self, js_cleaner, py_cleaner):
...
 
+        @expected_failing_versions(['3.6'])
         @transforms(
             js_bool = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 11]
```diff
@@ -870,6 +872,7 @@ def test_left_adjust(self, js_cleaner, py_cleaner):
...
             decimal = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 12]
```diff
@@ -898,6 +901,7 @@ def test_plus_sign(self, js_cleaner, py_cleaner):
 
             self.assertCodeExecution(tests, js_cleaner = js_cleaner, py_cleaner = py_cleaner)
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 55: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 57: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 80: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 82: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 13]
```diff
@@ -898,6 +901,7 @@ def test_plus_sign(self, js_cleaner, py_cleaner):
...
 
+        @expected_failing_versions(['3.6'])
         @transforms(
             js_bool = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 14]
```diff
@@ -898,6 +901,7 @@ def test_plus_sign(self, js_cleaner, py_cleaner):
...
             decimal = False,
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 20: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 22: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 15]
```diff
@@ -1293,6 +1297,7 @@ def test_signs_with_str(self):
 
         self.assertCodeExecution(test_str)
 
+    @expected_failing_versions(['3.6'])
     @transforms(decimal = False,)
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 24: (E251) unexpected spaces around keyword / parameter equals

Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 26: (E251) unexpected spaces around keyword / parameter equals

File: tests/datatypes/test_str.py [hunk 16]
```diff
@@ -1310,6 +1315,7 @@ def test_groupings(self, js_cleaner, py_cleaner):
         self.assertCodeExecution(test_str, js_cleaner=js_cleaner,
                                     py_cleaner=py_cleaner)
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 37: (E127) continuation line over-indented for visual indent

File: tests/builtins/test_pow.py [hunk 17]
```diff
@@ -1,6 +1,6 @@
 from unittest import expectedFailure
```
Comment (brutusthebee) [inline tests/builtins/test_pow.py]: At column 1: (F401) 'unittest.expectedFailure' imported but unused

File: tests/builtins/test_pow.py [hunk 18]
```diff
@@ -1,6 +1,6 @@
...
 
-from .. utils import TranspileTestCase, BuiltinFunctionTestCase, BuiltinTwoargFunctionTestCase
+from ..utils import TranspileTestCase, BuiltinFunctionTestCase, BuiltinTwoargFunctionTestCase, expected_failing_versions
```
Comment (brutusthebee) [inline tests/builtins/test_pow.py]: At column 120: (E501) line too long (120 > 119 characters)

File: tests/datatypes/test_str.py [hunk 19]
```diff
@@ -896,13 +899,14 @@ def test_plus_sign(self, js_cleaner, py_cleaner):
                 ]
```
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 17: (E124) closing bracket does not match visual indentation

File: tests/datatypes/test_str.py (hunk 16 above)
Comment (brutusthebee) [inline tests/datatypes/test_str.py]: At column 37: (E127) continuation line over-indented for visual indent

File: tests/modules/test_stdlib.py [hunk 20]
```diff
@@ -1,16 +1,16 @@
-from ..utils import TranspileTestCase
-
-import unittest
+from ..utils import TranspileTestCase, expected_failing_versions
```
Comment (brutusthebee) [inline tests/modules/test_stdlib.py]: At column 1: (F401) '..utils.expected_failing_versions' imported but unused

**Comment by martica** [comment]
There are still a couple unexpected successes that failed locally for me, and one test failure in the 3.6 run, but looking at the error, it seems a candidate for flakiness as it depends on set ordering.

**Comment by martica** [comment]
Everything that fails is now marked as expected.

**Comment by danyeaw** [comment]
These changes look great to help use get to a state where the expected tests are passing. If we get this merged in before #772, then I can remove the Python 3.4 expected failures since we are dropping support for that version.

**Comment by martica** [comment]
Cool. I'm looking forward to rebasing the rest of my changes on this and seeing everything green. :)

Is there anything else that needs to be done to merge this? Brutus made some style comments that I corrected. All but the first two are marked as outdated, but even the first two have been corrected, so I'm not sure why it isn't updating to outdated.

**Comment by danyeaw** [comment]
@martica Thanks! :heart:

**Pull request merged**